iceoryx2-bb/container/src/byte_string.rs

```diff
@@ -73,6 +74,28 @@ pub struct FixedSizeByteString<const CAPACITY: usize> {
 unsafe impl<const CAPACITY: usize> Send for FixedSizeByteString<CAPACITY> {}
 unsafe impl<const CAPACITY: usize> Sync for FixedSizeByteString<CAPACITY> {}
 
+impl<const CAPACITY: usize, const CAPACITY_OTHER: usize>
+    PartialOrd<FixedSizeByteString<CAPACITY_OTHER>> for FixedSizeByteString<CAPACITY>
+{
+    fn partial_cmp(&self, other: &FixedSizeByteString<CAPACITY_OTHER>) -> Option<Ordering> {
+        for i in 0..std::cmp::min(self.len, other.len) {
+            match unsafe { *self.data[i].as_ptr() }.cmp(unsafe { &*other.data[i].as_ptr() }) {
+                Ordering::Less => return Some(Ordering::Less),
+                Ordering::Greater => return Some(Ordering::Greater),
+                _ => (),
+            }
+        }
+
+        Some(self.len.cmp(&other.len))
```

Comment:
My C++ reviews earned me the name BracketMan, with my Rust reviews it will become IteratorMan :laughing: 

Please consider this
```rs
self.data[0..self.len]
    .iter()
    .zip(other.data[0..other.len].iter())
    .map(|(lhs, rhs)| unsafe { lhs.assume_init_read().cmp(rhs.assume_init_ref()) })
    .filter(|&ord| ord != Ordering::Equal)
    .next()
    .unwrap_or(self.len.cmp(&other.len))
```

cc @orecham might be also interesting for you